### PR TITLE
AC-544: Re-developing OpenMRSLogger

### PR DESCRIPTION
**Pull request by deepak140596**
## Description of what I changed

Added AsyncTask for saving the logs to file.

## Issue I worked on

JIRA Issue: https://issues.openmrs.org/browse/AC-544

## Checklist: I completed these to help reviewers :)

- [x ] My pull request only contains **ONE single commit**
(the number above, next to the 'Commits' tab is 1).

- [ ] I have **added tests** to cover my changes. (If you refactored
existing code that was well tested you do not have to add tests)

- [ ] All new and existing **tests passed**.

- [x ] My pull request is **based on the latest changes** of the master branch.

### CODE REVIEW

**Comment by deepak140596** [comment]
Hi @f4ww4z , 
Please do not merge this pull request now even if everything is right.
I would like some feedbacks on this.

I didn't use any other libraries as all the functions that those libraries provided was already present in the OpenMRSLogger. Just made sure that the saving to file was done asynchronously.

In this PR itself, I would like to format the Logs neatly and will make them ready for Analytics (we can save logs/messages in a constructive way for analytical reason , which is one of the main aspects of this app). I will add some functions that will help in formatting and adding some data to the log alongwith Logs.

File: openmrs-client/src/main/java/org/openmrs/mobile/application/OpenMRSLogger.java [hunk 1]
```diff
@@ -115,13 +121,22 @@ private static void saveToFile() {
...
             rotateLogFile();
         }
     }
 
+    private static  class SaveAsyncToFile extends AsyncTask<Void,Void,Void>{
+
+        @Override
+        protected Void doInBackground(Void... voids) {
+            saveToFileAsync();
+            return null;
```
Comment (VANKINEENITAWRUN) [inline openmrs-client/src/main/java/org/openmrs/mobile/application/OpenMRSLogger.java]: @deepak140596 you can remove `return null` statement.

**Comment by VANKINEENITAWRUN** [comment]
@deepak140596 Thanks for your PR. Looks good. I think we can merge and you can make new issue for adding Analytics. Please request a code review in jira. As per PR guidelines.

File: openmrs-client/src/main/java/org/openmrs/mobile/application/OpenMRSLogger.java (hunk 1 above)
Comment (deepak140596) [reply openmrs-client/src/main/java/org/openmrs/mobile/application/OpenMRSLogger.java]: Thanks for the review Tawrun,
we have to return null in the doInBackground() to goto the onPostExecute().

File: openmrs-client/src/main/java/org/openmrs/mobile/application/OpenMRSLogger.java [hunk 2]
```diff
@@ -115,7 +141,46 @@ private static void saveToFile() {
...
+        @Override
+        // custom message as string parameter
+        protected Void doInBackground(String... strings) {
+            String msg = strings[0];
+            writeMsgToFileAsync(msg);
+            return null;
+        }
+    }
+
+    private static void writeMsgToFileAsync(String msg){
```
Comment (deepak140596) [inline openmrs-client/src/main/java/org/openmrs/mobile/application/OpenMRSLogger.java]: This function writes custom messages to Log file

**Comment by deepak140596** [comment]
@VANKINEENITAWRUN ,I am very sorry as  I am closing and reopening this PR.
Record the link of his PR Review as it will help you with your GSOC Proposal.

**Pull request closed without merging**